### PR TITLE
React-redux boilerplate

### PR DESCRIPTION
**Pull request by masalcedo88**
- Chore: installs redux, react-redux and redux-thunk via npm
- Chore: wire up redux-devtools @ src/index.js
- Create: redux store @ src/index.js
- Create: dummy reducer @ reducers/index.js
- Refactors: moved the BrowserRouter over to base/App.js to avoid stacking since I needed to wrap App with redux Provider

Note: just checked the redux devtools extension and it's working properly on my localhost

EDIT 2:
- Refactor: separated store function and logic to src/store as example from redux github, now chooses between dev and prod envs.
- Create: src/api.js - placeholder file to showcase RESTful example for actions creators.
- Create: src/actions/types.js variables to avoid typos when inputing action type strings
- Create: src/actions/index.js example functions for fetching events and delete an event.
- Create: src/reducers/ boilerplate reducers for all data models with import @ reducers/index.js

Note: I think we need to discuss which data type for each model we're going to be using. (**For now** everything is a JS Object).

EDIT 3:
- Refactor:
- - rename src/api/huoa.js -> src/apis/content.js
- - rename src/actions/types.js -> src/actions/actionTypes.js
- - changes if/else -> ternary expression @ src/store/configureStore

### CODE REVIEW

File: src/api/huoa.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import axios from 'axios';
```
Comment (mattlaguardia) [inline src/api/huoa.js:1]: I dont love this file's naming convention 

`api.js` would be a better name

File: src/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import huoa from '../api/huoa';
```
Comment (mattlaguardia) [inline src/actions/index.js:1]: lets change this to be a bit more agnostic something more like 
```
import api from '../base/api';
```

File: src/store/configureStore.dev.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { createStore, applyMiddleware, compose } from 'redux';
+import thunk from 'redux-thunk';
+import reducers from '../reducers';
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const configureStore = () => {
+	const store = createStore(reducers, composeEnhancers(applyMiddleware(thunk)));
+	return store;
+};
```
Comment (mattlaguardia) [inline src/store/configureStore.dev.js:9]:
```suggestion
const configureStore = () => {
	return createStore(reducers, composeEnhancers(applyMiddleware(thunk)));
};
```

saves some bites in memory

File: src/actions/types.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+//Donations
```
Comment (mattlaguardia) [inline src/actions/types.js:1]: I feel like this file name should be action-types or in a directory called constants

File: src/store/configureStore.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+if (process.env.NODE_ENV === 'production') {
+	module.exports = require('./configureStore.prod');
+} else {
+	module.exports = require('./configureStore.dev');
+}
```
Comment (mattlaguardia) [inline src/store/configureStore.js:5]:
```suggestion
module.exports = process.env.NODE_ENV === 'production' ? 
  require('./configureStore.prod') : 
  require('./configureStore.dev');
```

File: src/store/configureStore.dev.js (hunk 3 above)
Comment (masalcedo88) [reply src/store/configureStore.dev.js]: done!

File: src/api/huoa.js (hunk 1 above)
Comment (masalcedo88) [reply src/api/huoa.js]: done!

File: src/actions/index.js (hunk 2 above)
Comment (masalcedo88) [reply src/actions/index.js]: done > src/apis/content.js

File: src/actions/types.js (hunk 4 above)
Comment (masalcedo88) [reply src/actions/types.js]: done src/actions/actionTypes.js

File: src/store/configureStore.js (hunk 5 above)
Comment (masalcedo88) [reply src/store/configureStore.js]: done!

**Pull request merged**